code/modules/mob/living/simple_animal/slime/powers.dm

```diff
@@ -224,3 +225,57 @@
 /datum/action/innate/slime/reproduce/Activate()
 	var/mob/living/simple_animal/slime/S = owner
 	S.Reproduce()
+
+/mob/living/simple_animal/slime/proc/make_baby(drop_loc, new_adult, new_nutrition, new_powerlevel, force_original_colour=FALSE, step_away=TRUE,datum/component/nanites/original_nanites=null)
+	var/child_colour = colour
+	if(!force_original_colour)
+		if(prob(mutation_chance))
+			if(transformeffects & SLIME_EFFECT_PYRITE)
+				slime_mutation = mutation_table(pick(slime_colours - list("rainbow")))
+			else if(mutation_chance >= 100)
+				child_colour = "rainbow"
+			child_colour = slime_mutation[rand(1,4)]
+		else
+			child_colour = colour
+	var/mob/living/simple_animal/slime/M = new(drop_loc, child_colour, new_adult)
+	M.transformeffects = transformeffects
+	M.effectsapplied = effectsapplied
+	if(ckey || transformeffects & SLIME_EFFECT_CERULEAN)
+		M.set_nutrition(new_nutrition) //Player slimes are more robust at spliting. Once an oversight of poor copypasta, now a feature!
+	M.powerlevel = new_powerlevel
+	if(transformeffects & SLIME_EFFECT_METAL)
+		M.maxHealth = round(M.maxHealth * 1.3)
+		M.health = M.maxHealth
+	if(transformeffects & SLIME_EFFECT_PINK)
+		M.grant_language(/datum/language/common, TRUE, TRUE)
+		var/datum/language_holder/LH = get_language_holder()
+		LH.selected_language = /datum/language/common
```

Comment:
Is this not setting the language of the slime making the new slime, not the baby slime?